src/script/lua_api/l_mainmenu.cpp

```diff
@@ -328,82 +328,42 @@ int ModApiMainMenu::l_get_favorites(lua_State *L)
 			}
 		}
 
-		if (!server["version"].asString().empty()) {
-			lua_pushstring(L,"version");
-			std::string topush = server["version"].asString();
-			lua_pushstring(L,topush.c_str());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["proto_min"].asString().empty()) {
-			lua_pushstring(L,"proto_min");
-			lua_pushinteger(L, server["proto_min"].asInt());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["proto_max"].asString().empty()) {
-			lua_pushstring(L,"proto_max");
-			lua_pushinteger(L, server["proto_max"].asInt());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["password"].asString().empty()) {
-			lua_pushstring(L,"password");
-			lua_pushboolean(L, server["password"].asBool());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["creative"].asString().empty()) {
-			lua_pushstring(L,"creative");
-			lua_pushboolean(L, server["creative"].asBool());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["damage"].asString().empty()) {
-			lua_pushstring(L,"damage");
-			lua_pushboolean(L, server["damage"].asBool());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["pvp"].asString().empty()) {
-			lua_pushstring(L,"pvp");
-			lua_pushboolean(L, server["pvp"].asBool());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["description"].asString().empty()) {
-			lua_pushstring(L,"description");
-			std::string topush = server["description"].asString();
-			lua_pushstring(L,topush.c_str());
-			lua_settable(L, top_lvl2);
-		}
-
-		if (!server["name"].asString().empty()) {
-			lua_pushstring(L,"name");
-			std::string topush = server["name"].asString();
-			lua_pushstring(L,topush.c_str());
-			lua_settable(L, top_lvl2);
+		for (const auto& string_member : {"version", "description", "name", "address", "port"})
```

Comment:
please declare these explicitly on a separate line

Reply:
this defeats the purpose of shortening the code; if the arrays are only used at one place, why explicitly declare them?

Reply:
For readability?